src/php/Lang/Collections/HashMap/HashMapNodeInterface.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Lang\Collections\HashMap;
+
+use IteratorAggregate;
+
+/**
+ * @template TKey
+ * @template TValue
+ */
+interface HashMapNodeInterface extends IteratorAggregate
+{
+    /**
+     * @param TKey $key
+     * @param TValue $value
+     */
+    public function put(int $shift, int $hash, $key, $value, Box $addedLeaf): HashMapNodeInterface;
+
+    /**
+     * @param TKey $key
+     */
+    public function remove(int $shift, int $hash, $key): ?HashMapNodeInterface;
+
+    /**
+     * @template TDefault
+     *
+     * @param TKey $key
+     * @param ?TDefault $notFound
+     *
+     * @return TValue|TDefault
+     */
+    public function find(int $shift, int $hash, $key, $notFound);
```

Comment:
`null` is a not super good here, since the value of in a map can be `null` too (it is ok for the test). Additionally, this is only used internally as part of the HashMap.